### ISSUE TITLE
0.7.0-rc2 template_file: diffs didn't match during apply.

### ISSUE THREAD

**Issue by hngkr** [hashicorp/terraform#7244]
### Terraform Version
Terraform v0.7.0-rc2 (46a0709bba004d8b6e0eedad411270b3ae135a9e)

### Affected Resource(s)
- template_file

### Terraform Configuration Files
```hcl
provider "aws" {
  region = "eu-west-1"
}

resource "aws_api_gateway_rest_api" "terraform_template_rest_api" {
  name = "TerraformTemplateTest"
  description = "This is for testing a Terraform Template bug"
}

resource "template_file" "dummy" {
  template = "${file("${path.module}/dummy.tpl")}"
  vars {
    rest_api = "${aws_api_gateway_rest_api.terraform_template_rest_api.id}"
  }
}
```

### Output
https://gist.github.com/hngkr/f98472c788ac883086425170b65b3f0d

### Expected Behavior
Should have created the template_file resource

### Actual Behavior
Failed with: diffs didn't match during apply
  Mismatch reason: extra attributes: vars.rest_api

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
Running it a second time works.
Same construction has worked fine in 0.7.0-rc1 and earlier.

**Comment by jbardin** [comment]
This looks similar to what was solved in #7205, except with maps instead of sets.

I'd like to figure out what changes between rc1 and rc2 to trigger it though

**Comment by hngkr** [comment]
At this point of the evening, I'm certainly not clever enough to look thru' what #7205 does, but I just tried with master - and it's complaining about the same thing as above.

**Comment by jbardin** [comment]
This regression is a side effect of #7082.

**Comment by glasser** [comment]
Running into this one too.

**Comment by colemickens** [comment]
Was this auto-closed on accident?

**Comment by jbardin** [comment]
Hi @colemickens,

Nope, but the links between the issues are confusing, both because I mentioned the this issue in the code, and didn't change the PR message to include the "fixes" notation. Sorry about that. 

This was closed by #7249

**Comment by hngkr** [comment]
I'll just add that I just tried to rerun the script from the reported issue above. I used Terraform from the master branch, and I'm no longer getting the error message. Everything works beautifully. Thanks a lot!
